precompile/fee_config_manager.go

```diff
@@ -90,14 +95,31 @@ func (c *FeeConfigManagerConfig) Equal(s StatefulPrecompileConfig) bool {
 	if !ok {
 		return false
 	}
-	return c.UpgradeableConfig.Equal(&other.UpgradeableConfig) && c.AllowListConfig.Equal(&other.AllowListConfig)
+	eq := c.UpgradeableConfig.Equal(&other.UpgradeableConfig) && c.AllowListConfig.Equal(&other.AllowListConfig)
+	if !eq {
+		return false
+	}
+
+	if c.InitialFeeConfig == nil {
+		return other.InitialFeeConfig == nil
+	}
```

Comment:
Can we switch this to use a length check? Otherwise, I believe that this will differentiate between an empty map and a nil map.

Can we add a test to confirm the behavior for that case?

Reply:
this is not a map, it's a struct. were you looking at this one?
https://github.com/ava-labs/subnet-evm/blob/2d3a51702ba1bfe1c27dcb58d9d634e9c4dc9a4f/precompile/contract_native_minter.go#L101-L103

Reply:
Yes I was, my bad.